src/components/organisms/Common/AssetContainer/hooks.ts

```diff
@@ -164,11 +164,9 @@ export default (teamId?: string, allowDeletion?: boolean) => {
 
   useEffect(() => {
     return () => {
-      setTimeout(() => {
-        setSort(undefined);
-        setSearchTerm(undefined);
-        gqlCache.evict({ fieldName: "assets" });
-      }, 200);
+      setSort(undefined);
+      setSearchTerm(undefined);
+      gqlCache.evict({ fieldName: "assets" });
```

Comment:
[imo] Since I'm not sure the reason you used `setTImeout` but if you still want to use it,

[must] you should make it like

```
useEffect(() => {
    const timer = 
      setTimeout(() => {
        setSort(undefined);
        setSearchTerm(undefined);
        gqlCache.evict({ fieldName: "assets" });
      }, 200);
      setSort(undefined);
      setSearchTerm(undefined);
      gqlCache.evict({ fieldName: "assets" });
	return clearTimeout(timer);
```

Reply:
Does Baba's code clears cache on UNMOUNTING the component? I think it does not work..

Reply:
I tested it just in case, but no it doesn't. So I will not use this